### PR TITLE
feat(metrics): Query tags across metrics

### PR DESCRIPTION
**Pull request by jjbayer**
Update the metrics API (still serving mock data) to the following, more RESTy url schema.
This also allows getting tag names & values without having to specify a metric.

```
/metrics/meta/
/metrics/meta/<metric-name>/
/metrics/tags/
/metrics/tags/<tag-name>/
/metrics/data/
```

### CODE REVIEW

File: src/sentry/api/endpoints/project_metrics.py [hunk 1]
```diff
@@ -1,36 +1,70 @@
...
-
-        if not features.has("organizations:metrics", project.organization, actor=request.user):
-            return Response(status=404)
-
         metrics = DATA_SOURCE.get_metrics(project)
         return Response(metrics, status=200)
 
 
+class ProjectMetricDetailsEndpoint(ProjectEndpoint):
+    """ Get metric name, type, unit and tag names """
```
[the remark this answers is not in the document]
Comment (priscilawebdev) [reply src/sentry/api/endpoints/project_metrics.py:18]: maybe we can mention project here too

**Review by priscilawebdev — APPROVED** [review]
I tested the API on the frontend and everything worked fine

File: src/sentry/api/endpoints/project_metrics.py [hunk 2]
```diff
@@ -1,36 +1,70 @@
...
-
         metrics = DATA_SOURCE.get_metrics(project)
         return Response(metrics, status=200)
 
 
+class ProjectMetricDetailsEndpoint(ProjectEndpoint):
+    """ Get metric name, available operations, metric unit and available tags """
+
+    def get(self, request, project, metric_name):
+        try:
```
Comment (markstory) [inline src/sentry/api/endpoints/project_metrics.py:21]: Do these endpoints need feature flag checks? Right now they look open to any organization.

File: src/sentry/api/endpoints/project_metrics.py [hunk 3]
```diff
@@ -1,36 +1,70 @@
...
+
+    def get(self, request, project, tag_name):
+
+        metric_names = request.GET.getlist("metric") or None
+
+        try:
+            tag_values = DATA_SOURCE.get_tag_values(project, tag_name, metric_names)
+        except InvalidParams as exc:
+            msg = str(exc)
+            if "Unknown tag" in msg:
```
Comment (markstory) [inline src/sentry/api/endpoints/project_metrics.py:63]: When this endpoint has real data would this be a separate error type?

File: src/sentry/api/endpoints/project_metrics.py [hunk 4]
```diff
@@ -43,9 +77,6 @@ class ProjectMetricsDataEndpoint(ProjectEndpoint):
 
     def get(self, request, project):
 
-        if not features.has("organizations:metrics", project.organization, actor=request.user):
```
Comment (markstory) [inline src/sentry/api/endpoints/project_metrics.py:46]: Don't you want this check to keep users from orgs that don't have access to metrics out of the endpoints?

File: src/sentry/api/endpoints/project_metrics.py (hunk 3 above)
Comment (jjbayer) [reply src/sentry/api/endpoints/project_metrics.py:82]: Yes, I've added a comment so we don't forget.

File: src/sentry/api/endpoints/project_metrics.py (hunk 4 above)
Comment (jjbayer) [reply src/sentry/api/endpoints/project_metrics.py]: @markstory Yes, thank you for noticing! Apparently I dropped these checks in a merge-fest and failed to restore them afterwards. Added them now.

**Review by markstory — APPROVED** [review]

**Pull request merged**